### ISSUE TITLE
[VRR] Direct Scanout causes stuttering

### ISSUE THREAD

**Issue by GrabbenD** [swaywm/sway#7714]
- **Sway Versions:**
   Stable: Sway 1.8.1 + Wlroots 1.6.2
   Next: Sway 1.9.0-dev-[363c579](https://github.com/swaywm/sway/commit/363c57984d08ff54bbf31f567ffcd4addad98753) + Wlroots 1.7.0-dev-[5f691259](https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/5f6912595e922c30beaf99190634bd1747be8f87)
   (Tested on Arch Linux with 23.1.6)

- **Debug Log:**
   Result from starting Sway 1.8.1 > Steam > Halo Infinite > Custom Game > Create Match > Play:
   https://gist.github.com/GrabbenD/58fdff75c802311036320393d03b7d0c

- **Configuration File:**
   Default, only tweaks are:
   ```
   output * {
       mode 3840x1600@119.982Hz
       adaptive_sync on
   }
   
   input type:pointer {
       accel_profile "flat"
       pointer_accel 0
   }
   ```

- **Description:**
  In fast paced games such as Halo Infinite (XWayland) textures looks like they're vibrating/fuzzy when moving around quickly. Furthermore the mouse movement feels very choppy/stuttery. This is very apparent when the FPS is below monitor's refresh rate but it can also be observed when the FPS is the same as the refresh rate. 

  In other words, VRR/Variable-Refresh-Rate/Freesync/Adaptive-Sync doesn't work properly.

- **Workaround:**
   **This can be completely fixed by starting Sway with: `$ sway -Dnoscanout`**

**Update:** see comment [below](https://github.com/swaywm/sway/issues/7714#issuecomment-1694356639)

**Comment by GrabbenD** [comment]
This issue is duplicate, here's a workaround:
https://github.com/swaywm/sway/issues/7370#issuecomment-1688612523

**This is most likely causing most of the stuttering on my end:**
https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3721

Other than that here's some related issues which can cause stuttering in Sway:
- Cursor:
   - https://github.com/swaywm/sway/issues/6832   
   - https://gitlab.freedesktop.org/drm/amd/-/issues/2186
   - https://github.com/swaywm/sway/issues/4763
   - https://github.com/samhh/dotfiles/issues/290
   - https://github.com/swaywm/sway/issues/5591
- DRM:
   - https://github.com/swaywm/wlroots/issues/894
   - https://gitlab.freedesktop.org/mesa/mesa/-/issues/4308
- VRR:
   - https://github.com/swaywm/sway/issues/5601
   - https://gitlab.freedesktop.org/wayland/wayland/-/issues/84#note_609628
   - https://github.com/swaywm/sway/issues/5076
- AMDGPU:
   - https://gitlab.freedesktop.org/drm/amd/-/issues/1500#note_1854170
   - https://gitlab.freedesktop.org/drm/amd/-/issues/2657
   - https://gitlab.freedesktop.org/drm/amd/-/issues/2460
- vblank:
   - https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/65
   - https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/3871
- MESA:
   - https://gitlab.freedesktop.org/mesa/mesa/-/issues/9189
   - https://gitlab.freedesktop.org/mesa/mesa/-/issues/9443

**Comment by GrabbenD** [comment]
I'm re-opening this to bring more attention to VRR not working properly in Sway as I've come to notice other people are experiencing this from reading various forum posts and issue reports.

**Comment by Nama** [comment]
> Update: I'm no longer observing this issue 

can't confirm.
Made a post
https://gitlab.freedesktop.org/drm/amd/-/issues/3166#note_2522283
